ral-facilities/operationsgateway
# DSEGOG-213 Adding @yarnpkg/sdks dependency for automatic SDKs installation

## Description
I was getting annoyed at having to run an extra command to get set up with the required SDKs for developing OperationsGateway while using Yarn 3 with PnP. We've got this documented in the repo wiki, which requires the user to run something like
```
yarn dlx @yarnpkg/sdks vscode
```
if they want to be able to resolve external dependencies in VSCode, for example. This is easily missed; I've done it myself several times and I wrote the damn thing.

There is a solution to make this happen automatically and this PR provides it.

First, we can add @yarnpkg/sdks to our `devDependencies`. What this does is it means we now only need to run the command `yarn sdks` and yarn will automatically install the correct SDKs for our environment. Another advantage of this is that renovate can now keep our SDKs up-to-date (in theory).

Second, now that we have this more generic command, we can add this to the `yarn postinstall` command to have it automatically run whenever the user runs `yarn install`. More users are aware of and should know to run `yarn install` before running `yarn start` so this improves the process overall.

## Testing instructions
The only way to properly test this is to clone a fresh copy of the repo in a brand new place on your computer. This eliminates the chance of any up-to-date lingering SDKs or Typescript libraries being resolved to the new workspace. Now open the workspace in your IDE of choice. If you're like me, the workspace will resolve the global version of Typescript I have installed on my computer (4.6.4 when we want the latest 4.9.3) and you won't be able to resolve external dependencies in the IDE.

Now run `yarn install`. This should set up the workspace with its normal dependencies plus its additional SDK files. Following this command's execution, your IDE _should_ now be able to resolve its external dependencies. If you're running VSCode, you should get a popup asking you to switch to the workspace version of Typescript (the newly installed 4.9.3), otherwise I believe this also happens automatically.


## Agile board tracking
Closes DSEGOG-213
